File: src/main/java/com/nhan/minisocial/core/repository/CommentRepository.java
package com.nhan.minisocial.core.repository;

import com.nhan.minisocial.core.entity.Comment;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CommentRepository extends JpaRepository<Comment, Long> {
    List<Comment> findCommentsByArticle(long articleId);
}

File: services/interaction-service-dao/src/main/kotlin/com/nhan/social/interaction/repository/CommentRepository.kt
package com.nhan.social.interaction.repository

import com.nhan.social.interaction.entity.Comment
import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepositoryBase
import io.quarkus.panache.common.Page
import io.quarkus.panache.common.Sort
import jakarta.enterprise.context.ApplicationScoped
import java.util.UUID

@ApplicationScoped
class CommentRepository : PanacheRepositoryBase<Comment, UUID> {

    fun findByTarget(targetId: UUID, targetType: String, page: Int, size: Int): List<Comment> =
        find(
            "targetId = ?1 and targetType = ?2 and visible = true",
            Sort.by("createdAt").ascending(),
            targetId, targetType,
        ).page(Page.of(page, size)).list()

    fun countByTarget(targetId: UUID, targetType: String): Long =
        count("targetId = ?1 and targetType = ?2 and visible = true", targetId, targetType)
}
